snowpack/src/scan-imports.ts

```diff
@@ -156,10 +157,21 @@ function parseFileForInstallTargets({
   contents,
 }: SnowpackSourceFile): InstallTarget[] {
   try {
-    if (baseExt === '.css' || baseExt === '.scss' || baseExt === '.sass' || baseExt === '.less') {
-      return parseCssForInstallTargets(contents);
-    } else {
-      return parseJsForInstallTargets(contents);
+    switch (baseExt) {
+      case '.css':
+      case '.less':
+      case '.sass':
+      case '.scss': {
+        return parseCssForInstallTargets(contents);
+      }
+      case '.html':
+      case '.svelte':
+      case '.vue': {
+        return parseJsForInstallTargets(extractJSFromHTML({contents, baseExt}));
+      }
+      default: {
+        return parseJsForInstallTargets(contents);
```

Comment:
Here's the default that introduced the bug with binary files.

Reply:
This introduced the `default`, but I think the code that was there before (the `if / else`) operated the same way? We still had `parseJsForInstallTargets` as the fallback case before. However, we weren’t always letting binary files go through this flow.